Sources/PrimerSDK/Classes/Core/BIN Data/CardValidationService.swift

```diff
@@ -0,0 +1,126 @@
+        let sanitizedCardNumber = cardNumber.replacingOccurrences(of: " ", with: "")
+        guard !sanitizedCardNumber.isEmpty, sanitizedCardNumber != mostRecentCardNumber else {
+            return
+        }
+        mostRecentCardNumber = sanitizedCardNumber
+        
+        let cardState = PrimerCardState(cardNumber: sanitizedCardNumber)
+        guard sanitizedCardNumber.count >= 6 else {
+            useLocalValidation(withCardState: cardState)
+            return
+        }
+        
+        debouncer.debounce { [weak self] in
+            self?.useRemoteValidation(withCardState: cardState)
+        }
+    }
+    
+    private func useRemoteValidation(withCardState cardState: PrimerCardState) {
+        delegate?.primerRawDataManager?(rawDataManager,
+                                        willFetchCardMetadataForState: cardState)
+        
+        let rawDataManager = rawDataManager
+        
+        _ = listCardNetworks(cardState.cardNumber).done { [weak self] result in
+            guard result.networks.count > 0 else {
+                self?.useLocalValidation(withCardState: cardState)
+                return
+            }
+            let cardMetadata = PrimerCardMetadata(availableCardNetworks: result.networks.map { network in
+                PrimerCardNetwork(displayName: network.displayName, networkIdentifier: network.value)
+            })
+            
+            self?.delegate?.primerRawDataManager?(rawDataManager,
+                                                  didReceiveCardMetadata: cardMetadata,
+                                                  forCardState: cardState)
+        }.catch { error in
+            // JN TODO: CHKT-1772 - send event
+            self.logger.error(message: "Error occurred while remotely validation '\(cardState.cardNumber)': \(error.localizedDescription)")
+            self.useLocalValidation(withCardState: cardState)
+        }
+    }
+    
+    func useLocalValidation(withCardState cardState: PrimerCardState) {
+        let localValidationNetwork = CardNetwork(cardNumber: cardState.cardNumber)
+        let displayName = localValidationNetwork.validation?.niceType ?? localValidationNetwork.rawValue.lowercased().capitalized
+        let cardNetwork = PrimerCardNetwork(displayName: displayName,
+                                            networkIdentifier: localValidationNetwork.rawValue)
+        delegate?.primerRawDataManager?(rawDataManager,
+                                        didReceiveCardMetadata: .init(availableCardNetworks: [cardNetwork]),
+                                        forCardState: cardState)
+    }
+    
+    // MARK: API Logic
+    
+    var validateCardNetworksCancellable: PrimerCancellable?
+    
+    private func listCardNetworks(_ cardNumber: String) -> Promise<Response.Body.Bin.Networks> {
+        
+        // ⚠️ We must only ever send eight or less digits to the endpoint
+        let cardNumber = String(cardNumber.prefix(8))
```

Comment:
maybe extract this number 8 to some constants? don't bother doing it if you will never have repetition of this same number in same context

Reply:
Good shout - I'll do this as part of CHKT-1885 (Modify card entry logic to account for eight digit constraint)